tasks/get-clair-results.yaml

```diff
@@ -0,0 +1,72 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: get-clair-scan
+spec:
+  params:
+    - name: sha
+      description: sha of an image
+      type: string
+    - name: pullspec
+      description: image full name with a tag
+      type: string
+  steps:
+    - name: get-vulnerabilities
+      image: quay.io/redhat-appstudio/hacbs-test:latest
+      script: |
+        #!/usr/bin/env bash
+        temp='$(params.sha)'
+        modifiedSha=${temp//:/%3A}
+        
+        #ugly safety mechanism - need to be improved in future, bleh :(
+        
+        function safety_mechanism {
+
+        #curl request
+        http_code=$(curl -o $(workspaces.clair-ws.path)/clair-result.json -w '%{http_code}' -H "Content-type: application/json" -XGET https://quay.io/api/v1/repository/$(echo '$(params.pullspec)' | sed "s/\(.*\):.*/\1/")/manifest/$modifiedSha/security?vulnerabilities=true)
```

Comment:
trying with sha256:e8f23243be6f20d77e6412046f3422d464d99c56e9f5061fd63572a043fe2607 and quay.io/redhat-appstudio/application-service:sha-75a4c79, getting {"detail": "Requires authentication", "error_message": "Requires authentication", "error_type": "invalid_token", "title": "invalid_token", "type": "https://quay.io/api/v1/error/invalid_token", "status": 401}

Reply:
Curl can be obtained in case of public repo at the moment, there is a story regarding having users providing they token:
https://issues.redhat.com/browse/SVPI-62, after users can provide their token manually, we can also provide scans from private repos.